torchvision/prototype/datasets/_folder.py

```diff
@@ -0,0 +1,44 @@
+import io
+import os
+import os.path
+import pathlib
+from typing import Union, Tuple, List, Dict, Any
+
+from torch.utils.data import IterDataPipe
+from torch.utils.data.datapipes.iter import FileLister, Concater, FileLoader, Mapper
+
+__all__ = ["from_image_folder"]
+
+
+def _collate(
+    data: Tuple[str, io.IOBase],
+    *,
+    label: int,
+    category: str,
+) -> Dict[str, Any]:
+    path, image_file = data
+    return dict(
+        path=path,
+        image=image_file,
```

Comment:
nit: maybe calling this `image` might be misleading because the image hasn't been decoded yet?

Reply:
Valid question. IMO decoding should be an in-place operation and since in most cases there will be some decoding at some point, I feel like "image" is ok as key. Plus, otherwise the user would need to specify the name of the decoded feature together with the feature to be decoded, e.g.

```python
dataset.decode_sample(datasets.decoder.pil, ("image_file", "image")
```

Reply:
We can revisit this in the early future, so let's not block on this.